lib/rds-snapshot-export-pipeline-stack.ts

```diff
@@ -0,0 +1,185 @@
+  DB_AUTOMATED_SNAPSHOT_CREATED = "RDS-EVENT-0091"
+}
+
+export interface RdsSnapshotExportPipelineStackProps extends cdk.StackProps {
+  /**
+   * Name of the S3 bucket to which snapshot exports should be saved.
+   *
+   * NOTE: Bucket will be created if one does not already exist.
+   */
+  readonly s3BucketName: string;
+
+  /**
+   * Name of the database cluster whose snapshots the function supports exporting.
+   */
+  readonly dbName: string;
+};
+
+export class RdsSnapshotExportPipelineStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props: RdsSnapshotExportPipelineStackProps) {
+    super(scope, id, props);
+
+    const bucket = new Bucket(this, "SnapshotExportBucket", {
+      bucketName: props.s3BucketName,
+      removalPolicy: config.get('defaultremovalpolicy'),
+      autoDeleteObjects: true,
+      blockPublicAccess: BlockPublicAccess.BLOCK_ALL,
+    });
+
+    const snapshotExportTaskRole = new Role(this, "SnapshotExportTaskRole", {
+      assumedBy: new ServicePrincipal("export.rds.amazonaws.com"),
+      description: "Role used by RDS to perform snapshot exports to S3",
+      inlinePolicies: {
+        "SnapshotExportTaskPolicy": PolicyDocument.fromJson({
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Action": [
+                "s3:PutObject*",
+                "s3:ListBucket",
+                "s3:GetObject*",
+                "s3:DeleteObject*",
+                "s3:GetBucketLocation"
+              ],
+              "Resource": [
+                `${bucket.bucketArn}`,
+                `${bucket.bucketArn}/*`,
+              ],
+              "Effect": "Allow"
+            }
+          ],
+        })
+      }
+    });
+
+    const lambdaExecutionRole = new Role(this, "RdsSnapshotExporterLambdaExecutionRole", {
+      assumedBy: new ServicePrincipal("lambda.amazonaws.com"),
+      description: 'RdsSnapshotExportToS3 Lambda execution role for the "' + props.dbName + '" database.',
+      inlinePolicies: {
+        "SnapshotExporterLambdaPolicy": PolicyDocument.fromJson({
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Action": "rds:StartExportTask",
+              "Resource": "*",
+              "Effect": "Allow",
+            },
+            {
+              "Action": "iam:PassRole",
+              "Resource": [snapshotExportTaskRole.roleArn],
+              "Effect": "Allow",
+            }
+          ]
+        })
+      },
+      managedPolicies: [
+        ManagedPolicy.fromAwsManagedPolicyName("service-role/AWSLambdaBasicExecutionRole"),
+      ],
+    });
+
+    const snapshotExportEncryptionKey = new Key(this, "SnapshotExportEncryptionKey", {
+      alias: props.dbName + "-snapshot-exports",
+      policy: PolicyDocument.fromJson({
+        "Version": "2012-10-17",
+        "Statement": [
+          {
+            "Principal": {
+              "AWS": [
+                (new AccountRootPrincipal()).arn,
```

Comment:
Here and line 138 you are creating new AccountRootPrincipal object. Can this be a common object?

Reply:
Fixed